src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -851,6 +851,42 @@ public void ExtensionWarningEvent(string hubName, string functionName, string in
                 configurationJsonString, hubName, LocalAppName, LocalSlotName, ExtensionVersion);
         }
 
+        public void RetrievingToken(
+            string hubName,
+            string resource)
+        {
+            EtwEventSource.Instance.RetrievingToken(
+                hubName,
+                LocalAppName,
+                LocalSlotName,
+                resource,
+                ExtensionVersion);
+
+            this.logger.LogInformation(
+                "Attempting to retrieve authentication token for resource '{resource}'. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}.",
+                resource, hubName, LocalAppName, LocalSlotName, ExtensionVersion);
+        }
+
+        public void TokenRetrievalFailed(
+            string hubName,
+            string resource,
+            Exception exception)
+        {
+            EtwEventSource.Instance.TokenRetrievalFailed(
+                hubName,
+                LocalAppName,
+                LocalSlotName,
+                resource,
+                exception.Message,
+                ExtensionVersion);
+
+            this.logger.LogWarning(
```

Comment:
Should this be `LogError`? In EtwEventSource.TokenRetrievalFailed it's marked as an error, and it appears to be a critical issue.
```suggestion
            this.logger.LogError(
```

Reply:
Ah that makes sense. I'll make `TokenRetrievalFailed` critical (as it prevents processing) while `TokenRenewalFailed` can be an error.

Reply:
I think TokenRenewalFailed is fine as a warning since it doesn't necessarily result in any failures, but TokenRetrievalFailed as an error feels like the right call to me.